.drone.star

```diff
@@ -0,0 +1,150 @@
+def main(ctx):
+    platforms = [ "amd64","arm64"]
+
+    volumes = [
+        {
+            "name": "cache",
+            "path": "/go",
+        }]
+
+    workspace = {
+        "path": "/go/src/github.com/${DRONE_REPO}"
+    }
+
+    resources = {
+         "requests": {
+              "cpu": 400,
+              "memory": "2Gi"
+         }
+    }
+
+    trigger = { "branch": ["main"] }
+
+    test_steps = {
+        "test": append_volumes(test_step(), volumes),
+    }
+    
+    pipelines = [
+        {
+            "kind": "pipeline",
+            "type": "kubernetes",
+            "name": "pre-build",
+            "resources": resources,
+            "steps": [append_volumes(license_step(), volumes)],
+            "trigger": trigger,
+            "volumes": volumes
+        }
+    ]
+    for plat in platforms:
+        pipe = { 
+            "kind": "pipeline",
+            "type": "kubernetes",
+            "name": plat,
+            "platform": { "arch": plat },
+            "resources": resources,
+            "steps": [v for v in test_steps.values()],
```

Comment:
since test step doesn't change between arm vs amd, I think it should be part of pre-build

Reply:
I personally want to see the test suite pass on each arch. I suppose incompatible libraries would get caught in the build step, but tests still require compilation which is platform dependent.

Reply:
gotcha